committeeoversightapp/models.py

```diff
@@ -94,10 +88,208 @@ def chair(self):
     def hide_rating(self):
         return CommitteeDetailPage.objects.get(committee=self.id).hide_rating
 
+    @property
+    def ratings(self):
+        return self.committeerating_set.all().order_by('-congress')
+
+    @property
+    def ratings_reverse(self):
+        return self.committeerating_set.all().order_by('congress')
+
+    @property
+    def latest_rating(self):
+        return self.ratings[0]
+
+    @property
+    def max_chp_points(self):
+        return self.ratings.aggregate(Max('chp_points'))['chp_points__max']
+
+    @property
+    def investigative_oversight_hearings_max(self):
+        return self.get_max_rating('investigative_oversight_hearings')
+
+    @property
+    def policy_legislative_hearings_max(self):
+        return self.get_max_rating('policy_legislative_hearings')
+
+    @property
+    def total_hearings_max(self):
+        return self.get_max_rating('total_hearings')
+
+    @property
+    def investigative_oversight_hearings_avg(self):
+        return self.get_avg_rating('investigative_oversight_hearings')
+
+    @property
+    def policy_legislative_hearings_avg(self):
+        return self.get_avg_rating('policy_legislative_hearings')
+
+    @property
+    def total_hearings_avg(self):
+        return self.get_avg_rating('total_hearings')
+
+    def get_avg_rating(self, hearing_type):
+        return round(self.ratings \
+            .aggregate(
+                Avg(hearing_type)
+            )[hearing_type + '__avg'], 2)
+
+    def get_max_rating(self, hearing_type):
+        return round(self.ratings \
+            .aggregate(
+                Max(hearing_type)
+            )[hearing_type + '__max'], 2)
+
     def __str__(self):
         return self.name
 
 
+class Congress(models.Model):
+    start_date = models.DateField()
+    end_date = models.DateField()
+
+    @property
+    def label(self):
+        '''
+        116 => '116th Congress'
+        '''
+        return '{} Congress'.format(ordinal(self.id))
+
+    @property
+    def is_current(self):
+        if self.start_date < date.today() < self.end_date:
+            return True
+        else:
+            return False
+
+    @property
+    def percent_passed(self):
+        days_in_session = 668
+        days_passed = (date.today() - self.start_date).days
+        return round(days_passed * 100.0 / days_in_session, 2)
```

Comment:
It shouldn't ever be over 100! I'll clamp it. 

For your first question, I'm in the habit of putting `100.0` second because it converts `int`s to `floats`. For example:

```python
 >>> days_in_session = 668
>>> days_passed = 400
>>> days_passed / days_in_session * 100.0
0.0
>>> days_passed * 100.0 / days_in_session
59.880239520958085
```

Is there something else you'd recommend here?

Reply:
I did not actually know this, but apparently Python 3.x will always do floating point division, regardless of the types of the inputs. https://codereview.stackexchange.com/a/69544

```bash
(committee-oversight) call-me-hank:committee-oversight hannah$ python3
Python 3.7.3 (default, Mar 27 2019, 09:23:15)
[Clang 10.0.1 (clang-1001.0.46.3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> 3/5
0.6
```

You can do the future import to safeguard against people running the app in Python 2.x, though chances are slim thanks to our explicit Python version in our Dockerfile!

Reply:
Huh TIL! Thanks for digging into this.